src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -267,8 +267,13 @@ private fun inferArrayType(expr: RsArrayExpr): Ty {
     } else {
         val elements = expr.arrayElements
         if (elements.isNullOrEmpty()) return TySlice(TyUnknown)
+
+        var elementType: Ty = TyUnknown;
         // '!!' is safe here because we've just checked that elements isn't null
-        elements!![0].type to elements.size
+        for (e in elements!!) {
```

Comment:
Do you think it is OK if we calculate the type for all array/vec elements?
Or should be an upper bound (e.g. only look at the first 20 array elements, quit after the first full type?)